torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -845,7 +845,8 @@ Tensor* TensorExprKernel::computeValue(const torch::jit::Value* v) {
 
     case aten::relu: {
       return computeOneOperand("aten_relu", v, [](const ExprHandle& a) {
-        return Max::make(a, 0, false);
+        auto zero = Cast::make(a.dtype(), 0);
+        return ifThenElse(CompareSelect::make(a, zero, kLT), zero, a);
```

Comment:
Why not `Max::make`?

Reply:
It's a bit of a micro-optimization, since we know 0 is never NaN, we can avoid checking either argument by writing the comparison this way.  We might be able to optimize away the checks in Max::make, but it can't to do it up front :-p

Reply:
I actually prefer Max in this case. Because zero is a constant for our generated code, so the compiler is very good at figuring out isnan(0) in this case.